### PR TITLE
Optimised Queries

### PR DESCRIPTION
**Pull request by gunjanpatel**
@nadeeshaan Please provide us a testing Instructions or steps for your several commits.

### CODE REVIEW

**Comment by gunjanpatel** [comment]
# Please don't merge this pull request until we got it tested. Thanks.

@nadeeshaan Please text me before you send new commits, I need to talk with you. :smile:

File: administrator/components/com_categories/models/fields/categoryedit.php [hunk 1]
```diff
@@ -63,56 +63,58 @@ protected function getOptions()
 
 		$db = JFactory::getDbo();
 		$query = $db->getQuery(true)
-			->select('a.id AS value, a.title AS text, a.level, a.published')
-			->from('#__categories AS a')
-			->join('LEFT', $db->quoteName('#__categories') . ' AS b ON a.lft > b.lft AND a.rgt < b.rgt');
+			->select('DISTINCT a.id AS value, a.title AS text, a.level, a.published');
+		$subQuery = $db->getQuery(true)
```
Comment (gunjanpatel) [inline administrator/components/com_categories/models/fields/categoryedit.php:70]: Please add a blank line here. So, it looks nice and separated from above and not mix with the above.

**Comment by gunjanpatel** [comment]
I have added patch tester on our testing site. So, when you will open link http://gsoc.redhost.dk/administrator/index.php?option=com_patchtester&view=pulls you will find pull request and you will able to apply patch. 

**_Ask me if you want back-end access.**_

**Comment by alikon** [comment]
yes please send me backend access

**Comment by gunjanpatel** [comment]
Please check your email for back-end access.  @alikon It seems that you closed PR by mistake, I am reopen it.

Thanks. :)

File: administrator/components/com_categories/models/fields/categoryedit.php (hunk 1 above)
Comment (gunjanpatel) [inline administrator/components/com_categories/models/fields/categoryedit.php:70]: @nadeeshaan Please fix this line.

**Comment by gunjanpatel** [comment]
Test results from our testing site not looks promising.
add_new_article-before-after-pr-4

**Pull request closed without merging**